app/service/search_index.rb

```diff
@@ -0,0 +1,17 @@
+class SearchIndex
+  def self.add(record)
+    if search_index_callbacks_enabled?
```

Comment:
How about adding a logger step?

The other option is the one I mentioned before, which is running a "re-index all the things every night at 3am" rake task, just in case.

Which do you think makes more sense?